src/bmi/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+include(${PROJECT_SOURCE_DIR}/cmake/dynamic_sourced_library.cmake)
+dynamic_sourced_cxx_library(ngen_bmi "${CMAKE_CURRENT_SOURCE_DIR}")
+
+add_library(NGen::ngen_bmi ALIAS ngen_bmi)
+
+target_include_directories(ngen_bmi PUBLIC
+        ${PROJECT_SOURCE_DIR}/include/
+        ${PROJECT_SOURCE_DIR}/include/bmi
+        ${PROJECT_SOURCE_DIR}/include/geojson
+        ${PROJECT_SOURCE_DIR}/include/core/mediator
```

Comment:
Are all of these directories necessary?

Reply:
To piggyback off this, if these headers are needed, we should ideally be linking to the respective targets instead of including directories.

Reply:
I'll take a look through this again.  I know these two additions were required to get a clean build.

Reply:
@program-- I tried this out with the BMI lib subdir here, and it seems to work ok, but leads to more and more `ld warning` about duplicate libraries as well